vunit/test/unit/test_test_suites.py

```diff
@@ -105,5 +82,74 @@ def _read_test_results(contents, expected_test_cases):
                           config=None,
                           elaborate_only=False,
                           test_suite_name=None,
-                          test_cases=expected_test_cases)
-            return run._read_test_results(file_name=file_name)  # pylint: disable=protected-access
+                          test_cases=expected)
+            results = run._read_test_results(file_name=file_name)  # pylint: disable=protected-access
+            if do_assert:
+                self.assertEqual(results, expected)
+            return results
+
+    def test_exit_code(self):
+        """
+        Test that results are overwritten when none is FAILED but the exit code is nonzero
+        """
+
+        def test(contents, results, expected=None, werechecked=None):

```

Comment:
Terminology is not consistent. werechecked is sometimes called done.

Reply:
`werechecked` is a list of `waschecked`, which is the expected Boolean value of `done`. `done` is the actual value returned by function `_check_results`.

I could rename `werechecked` to `were_expected_to_be_checked` and `waschecked` to `was_expected_to_be_checked`; so `done` would be `was_actually_checked`.

For now, I changed it to ommit naming `done` explicitly, by comparing dicts at once, instad of using two asserts.